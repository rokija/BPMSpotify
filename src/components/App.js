import React from 'react';
import PropTypes from 'prop-types';
import { Switch, NavLink, Route } from 'react-router-dom';
import HomePage from './HomePage';
import NotFoundPage from './NotFoundPage';
import Callback from '../containers/Callback';
import LoginPage from '../containers/LoginPage';
import Search from '../components/Search';

class App extends React.Component {
  render() {
    const activeStyle = { color: 'blue' };
    return (
      <div className="base-container">
        <div className="header-container">
            <NavLink exact to="/" activeStyle={activeStyle}>Home</NavLink>
            {' | '}
            <NavLink to="/search" activeStyle={activeStyle}>Search</NavLink>
        </div>
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route exact path="/callback" component={Callback} />
          <Route path="/search" component={Search} />
          <Route path="/login" component={LoginPage} />
          <Route component={NotFoundPage} />
        </Switch>
      </div>
    );
  }
}

App.propTypes = {
    children: PropTypes.element
};

export default App;

