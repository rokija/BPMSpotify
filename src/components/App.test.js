import React from 'react';
import { shallow } from 'enzyme';
import App from './App';


describe('App', () => {
    const wrapper = shallow(<App />);

    it('renders properly', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
