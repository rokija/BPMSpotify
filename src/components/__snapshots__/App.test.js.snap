// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  className="base-container"
>
  <div
    className="header-container"
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/callback"
    />
    <Route
      component={[Function]}
      path="/search"
    />
    <Route
      component={[Function]}
      path="/login"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
